### PR TITLE
Add syntax highlighting for Java constants and class names when creating objects.

### PR DESCRIPTION
**Pull request by PerilousBooklet**

### CODE REVIEW

**Comment by PerilousBooklet** [comment]
I'm waiting for [this](https://github.com/lite-xl/lite-xl/issues/1746) to be solved to request a review.

File: plugins/language_java.lua [hunk 1]
```diff
@@ -6,19 +6,21 @@ syntax.add {
...
-    { pattern = "'\\x%x?%x?%x?%x'",         type = "string"   }, -- character hexadecimal escape sequence
-    { pattern = "'\\u%x%x%x%x'",            type = "string"   }, -- character unicode escape sequence
-    { pattern = "'\\?.'",                   type = "string"   }, -- character literal
-    { pattern = "-?0x%x+",              type = "number"   },
-    { pattern = "-?%d+[%d%.eE]*f?",     type = "number"   },
-    { pattern = "-?%.?%d+f?",           type = "number"   },
-    { pattern = "[%+%-=/%*%^%%<>!~|&]", type = "operator" },
-    { pattern = "[%a_][%w_]*%f[(]",     type = "function" },
-    { pattern = "[%a_][%w_]*",          type = "symbol"   },
+    { pattern = "//.-\n",                            type = "comment"  },
```
Comment (Guldoman) [inline plugins/language_java.lua:9]:
```suggestion
    { pattern = "//.*",                              type = "comment"  },
```

File: plugins/language_java.lua [hunk 2]
```diff
@@ -6,19 +6,21 @@ syntax.add {
...
+    { pattern = { "'", "'", '\\' },                  type = "string"   },
+    { pattern = "'\\x%x?%x?%x?%x'",                  type = "string"   }, -- character hexadecimal escape sequence
+    { pattern = "'\\u%x%x%x%x'",                     type = "string"   }, -- character unicode escape sequence
+    { pattern = "'\\?.'",                            type = "string"   }, -- character literal
+    { pattern = "-?0x%x+",                           type = "number"   },
+    { pattern = "-?%d+[%d%.eE]*f?",                  type = "number"   },
+    { pattern = "-?%.?%d+f?",                        type = "number"   },
+    { pattern = "[%+%-=/%*%^%%<>!~|&]",              type = "operator" },
+    { pattern = "[%a_][%w_]*%f[(]",                  type = "function" },
+    { regex   = "[A-Za-z]+(?=\\s[a-z0-9]+\\s\\=\\s[new])",   type = "function" }, -- Class name in class instance
```
Comment (Guldoman) [inline plugins/language_java.lua:21]: what is `[new]` for?

**Review by Guldoman — COMMENTED** [review]
Needs a version bump.

File: plugins/language_java.lua (hunk 2 above)
Comment (PerilousBooklet) [reply plugins/language_java.lua]: It was actually redundant.

File: plugins/language_java.lua [hunk 3]
```diff
@@ -6,19 +6,21 @@ syntax.add {
...
+    { pattern = { "'", "'", '\\' },                     type = "string"   },
+    { pattern = "'\\x%x?%x?%x?%x'",                     type = "string"   }, -- character hexadecimal escape sequence
+    { pattern = "'\\u%x%x%x%x'",                        type = "string"   }, -- character unicode escape sequence
+    { pattern = "'\\?.'",                               type = "string"   }, -- character literal
+    { pattern = "-?0x%x+",                              type = "number"   },
+    { pattern = "-?%d+[%d%.eE]*f?",                     type = "number"   },
+    { pattern = "-?%.?%d+f?",                           type = "number"   },
+    { pattern = "[%+%-=/%*%^%%<>!~|&]",                 type = "operator" },
+    { pattern = "[%a_][%w_]*%f[(]",                     type = "function" },
+    { regex   = "[A-Za-z]+(?=\\s[a-z0-9]+\\s\\=\\s)",   type = "function" }, -- Class name in class instance
```
Comment (Guldoman) [inline plugins/language_java.lua:21]: Why is this using a non-capturing group?

Comment (Guldoman) [reply plugins/language_java.lua:21]: Should it use `\s+` so that multiple spaces are still valid?

Comment (PerilousBooklet) [reply plugins/language_java.lua:21]: Now that you mention it, yes it should.

Comment (Guldoman) [reply plugins/language_java.lua:21]: Should it also allow for underscores in the middle of the words?

Comment (PerilousBooklet) [reply plugins/language_java.lua:21]: I've never seen class names with underscores.

Comment (Guldoman) [reply plugins/language_java.lua:21]: Yeah, but it's just a convention afaik.

Comment (PerilousBooklet) [reply plugins/language_java.lua:21]: Mmh, ok I'll add it.

**Comment by PerilousBooklet** [comment]
I was a bit rough with the commits.
I forgot to git pull the previous conflict-solve.

**Comment by PerilousBooklet** [comment]
OK, so this PR should be ready. The regex not-yet-implemented `look-behind` functionality is no longer necessary.

File: plugins/language_java.lua [hunk 4]
```diff
@@ -6,19 +6,21 @@ syntax.add {
...
+    { pattern = { "'", "'", '\\' },                             type = "string"   },
+    { pattern = "'\\x%x?%x?%x?%x'",                             type = "string"   }, -- character hexadecimal escape sequence
+    { pattern = "'\\u%x%x%x%x'",                                type = "string"   }, -- character unicode escape sequence
+    { pattern = "'\\?.'",                                       type = "string"   }, -- character literal
+    { pattern = "-?0x%x+",                                      type = "number"   },
+    { pattern = "-?%d+[%d%.eE]*f?",                             type = "number"   },
+    { pattern = "-?%.?%d+f?",                                   type = "number"   },
+    { pattern = "[%+%-=/%*%^%%<>!~|&]",                         type = "operator" },
+    { pattern = "[%a_][%w_]*%f[(]",                             type = "function" },
+    { regex   = "(?:\\w+\\.?)+(?=\\s+\\w++\\s+\\=\\s+)",        type = "function" }, -- Class name in class instance
```
Comment (Guldoman) [inline plugins/language_java.lua:21]: What is this supposed to match?

Comment (PerilousBooklet) [reply plugins/language_java.lua:21]: It's something that Eclipse does, it highlights the name of the class when creating an object.

Comment (PerilousBooklet) [reply plugins/language_java.lua:21]: image

Comment (Guldoman) [reply plugins/language_java.lua:21]: Then I guess you should make the spaces around `=` optional.

**Pull request merged**